Add docs and integration tests for Auto-compaction snapshot status API (#10510)
apache/druid: docs/operations/metrics.md

|`segment/dropQueue/count`|Number of segments to drop.|server.|Varies.|
|`segment/size`|Total size of used segments in a data source. Emitted only for data sources to which at least one used segment belongs.|dataSource.|Varies.|
|`segment/count`|Number of used segments belonging to a data source. Emitted only for data sources to which at least one used segment belongs.|dataSource.|< max|
|`segment/overShadowed/count`|Number of overshadowed segments.| |Varies.|
|`segment/unavailable/count`|Number of segments (not including replicas) left to load until segments that should be loaded in the cluster are available for queries.|dataSource.|0|
|`segment/underReplicated/count`|Number of segments (including replicas) left to load until segments that should be loaded in the cluster are available for queries.|tier, dataSource.|0|
|`tier/historical/count`|Number of available historical nodes in each tier.|tier.|Varies.|
|`tier/replication/factor`|Configured maximum replication factor in each tier.|tier.|Varies.|
|`tier/required/capacity`|Total capacity in bytes required in each tier.|tier.|Varies.|
|`tier/total/capacity`|Total capacity in bytes available in each tier.|tier.|Varies.|

> |`compact/task/count`|Number of task issued in the auto compaction run.| |Varies.|

Reviewer: ```suggestion
|`compact/task/count`|Number of tasks issued in the auto compaction run.| |Varies.|
```

Author: Done

> |`compactTask/maxSlot/count`|Number of max task slot that can be used for auto compaction tasks in the auto compaction run.| |Varies.|

Reviewer: ```suggestion
|`compactTask/maxSlot/count`|Max number of task slots that can be used for auto compaction tasks in the auto compaction run.| |Varies.|
```

Author: Done

> |`compactTask/availableSlot/count`|Number of available task slot that can be used for auto compaction tasks in the auto compaction run. (this is max slot minus any currently running compaction task)| |Varies.|

Reviewer: ```suggestion
|`compactTask/availableSlot/count`|Number of available task slots that can be used for auto compaction tasks in the auto compaction run. (this is max slot minus any currently running compaction task)| |Varies.|
```

Author: Done

> |`segment/waitCompact/bytes`|Total byte of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/waitCompact/bytes`|Total bytes of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

Author: Done

> |`segment/waitCompact/count`|Total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/waitCompact/count`|Total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

Author: Done

> |`interval/waitCompact/count`|Total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|

Reviewer: ```suggestion
|`interval/waitCompact/count`|Total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

Author: Done

> |`segment/compacted/bytes`|Total byte of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/compacted/bytes`|Total bytes of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

Author: Done

> |`segment/compacted/count`|Total number of segments of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/compacted/count`|Total number of segments of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

Author: Done

> |`interval/compacted/count`|Total number of intervals of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|

Reviewer: ```suggestion
|`interval/compacted/count`|Total number of intervals of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

Author: Done

> |`segment/skipCompact/bytes`|Total byte of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/skipCompact/bytes`|Total bytes of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

Author: Done

> |`segment/skipCompact/count`|Total number of segments of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|

Reviewer: ```suggestion
|`segment/skipCompact/count`|Total number of segments of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

Author: Done

> |`interval/skipCompact/count`|Total number of intervals of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|

Reviewer: ```suggestion
|`interval/skipCompact/count`|Total number of intervals of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

Author: Done
